### ISSUE TITLE
Flatpak Support

### ISSUE THREAD

**Issue by sebasmagri** [anbox/anbox#62]
Make sure you are running the latest version of Anbox before reporting an issue.

Please also check that no similar bug is already reported. Have a look on the
list of open bugs at https://github.com/anbox/anbox/issues

**Anbox release (`anbox version`) and system info (`anbox system-info`):**

**Description of the problem:**
I would like to have anbox available on FlatPak format. It has not been possible for me to use snap on ArchLinux.

**Expected:**
Be able to install Anbox with FlatPak

**Additional info:**

**Comment by jdsmit60** [comment]
If you really want to run anbox on Arch (like me) despite Arch not being supported, there's an AUR package: anbox-git. You'll have to install boost and doxygen before installing anbox-git, since those two packages are missing dependencies for a package that anbox-git pulls in.

**Comment by TingPing** [comment]
Based on the readme you can't Flatpak this in the current state:

> - Out-of-tree kernel modules for binder and ashmem as no distribution kernel ships both enabled.
> - A udev rule to set correct permissions for /dev/binder and /dev/ashmem
> - A upstart job which starts the Anbox session manager as part of a user session.

Flatpak's can't get root access and can't install things onto the host. They can't add udev rules (they aren't guaranteed to be portable between udev versions), and well upstart is ofc not remotely portable.

**Comment by intika** [comment]
what about https://code.puri.sm/Librem5/anbox-flatpak

**Comment by TingPing** [comment]
> The installation process currently consists of a few steps which will add additional components to your host system.
> To make this process as easy as possible we have bundled the necessary steps in a snap

To call that a Flatapk is some sort of joke.

**Comment by aviwad** [comment]
@intika that link isn't available anymore. Any flatpak progress?

**Comment by intika** [comment]
Here is what it was about http://web.archive.org/web/20180521070730/https://code.puri.sm/Librem5/anbox-flatpak but this is dead any way... i guess that this was not working in the first place, i did not test it nor have a copy

**Comment by PureTryOut** [comment]
Flatpak is way more supported than Snap is on any distribution I'm not sure why Snap is even used. Could Snap be dropped in favor of Flatpak maybe?

**Comment by rugk** [comment]
> Flatpak is way more supported than Snap is on any distribution I'm not sure why Snap is even used. 

To provide some reference/source/citation for that claim – because it is actually true – see [this comparison from 2018](https://kamikazow.wordpress.com/2018/06/08/adoption-of-flatpak-vs-snap-2018-edition/).

> Could Snap be dropped in favor of Flatpak maybe?

I guess that's technically hard, but if one has to made the decision, I would also argue in favor of that.

**Comment by DontUseGithub** [comment]
Ping. Any updates?

My OS doesn't have systemd so I can't install snaps, flatpaks work great tho.

**Comment by divansantana** [comment]
FYI Guix has this packaged now.

Guix is cross platform package management system similar to flatpak, snap, but more like nix.

guix install anbox .

**Comment by DontUseGithub** [comment]
> FYI Guix has this packaged now.
> 
> Guix is cross platform package management system similar to flatpak, snap, but more like nix.
> 
> guix install anbox .

Are you sure? I can't seem to find it here https://guix.gnu.org/packages/A/page/2/

**Comment by divansantana** [comment]
@DontUseGithub Sorry, it's not yet upstream. It's [here](https://gitlab.com/pkill-9/guix-packages-free/commit/8494650871ed9e5002de0d5c5559c83bdeb9a36b) at https://gitlab.com/pkill-9/guix-packages-free channel.

**Comment by damianatorrpm** [comment]
+1 for flatpak

**Comment by Nonononoki** [comment]
+1 for flatpak, which is superior to snaps

**Comment by PureTryOut** [comment]
Guys c'mon, Github has a thumbs-up feature for this, no need to bump the thread with +1.

**Comment by aspieln3r** [comment]
I believe anbox is missing a lot of userbase because the only way of distribution is snap. This is actually a cool piece of software and a flatpak or appimage will be great as snap is not supported by many linux distributions.

**Comment by CEHitchens** [comment]
Yes, I agree with aspieln3r regarding snap. Snap(d) runs with quite extensive permissions which I am not willing to grant. Flatpak, AppImage, PPA, something we could build ourselves, almost any alternative would receive a warmer welcome than a snap.

**Comment by Mhowser** [comment]
@morphis Could we get this issue pinned please?

**Comment by ShadowJonathan** [comment]
> @DontUseGithub Sorry, it's not yet upstream. It's [here](https://gitlab.com/pkill-9/guix-packages-free/commit/8494650871ed9e5002de0d5c5559c83bdeb9a36b) at https://gitlab.com/pkill-9/guix-packages-free channel.

I just talked to the maintainer of that channel, `anbox` doesn't want to build anymore, and so they marked it as inactive.

Official `guix` support would be nice, though, so far my experience with guix (as short as it is) has been relatively positive, so supporting it could instantly introduce a wider support for all kinds of distros, instead of locking it to Canonical & Derivatives via `snapd`, which (imo) has a questionable outlook on the platform by its developers.

**Comment by BrainBlasted** [comment]
> * Out-of-tree kernel modules for binder and ashmem as no distribution kernel ships both enabled.
> * A udev rule to set correct permissions for /dev/binder and /dev/ashmem
> * A upstart job which starts the Anbox session manager as part of a user session.

Are these issues still applicable?

**Comment by tjbrn** [comment]
any updates?

**Comment by jack9603301** [comment]
I suggest that flatpak be considered, because snap is being disabled by the community due to some problems (it will automatically override user packages), and for me, I would not use snap if I didn't have a choice. I prefer flatpak, apt and source install

**Comment by sonnyp** [comment]
> So to be useful at all, a hypothetical Anbox Flatpak will have to run with full host access (--talk-name=org.freedesktop.Flatpak/flatpak-spawn --host) and will only be able to use Flatpak as a distribution medium and it is very unlikely that this situation will change any time soon.

I think this issue is about distributing Anbox as a flatpak, sandboxing is nice to have but definitely not required to fix the typical problem of "I don't want to, or I can't use the snap".

Even Flathub doesn't have strict requirements with sandboxing ([ref](https://github.com/flathub/flathub/wiki/App-Requirements#permissions)). Considering the requirements here it's not clear if Flathub would accept it though, but it's possible to use a Github repository as flatpak repository and distribute it yourself without too much hassle.

**Comment by theo-brown** [comment]
Has anyone started work on this? Does anyone know how much of a challenge it would be?

**Comment by krillin666** [comment]
Any update on this ? :grin:

**Comment by ShadowJonathan** [comment]
From first glance, waydroid already looks much easier to set up, much more intuitive, and easier to integrate with non-Ubuntu distros, I'll try it this afternoon.

Also, "it uses wayland" is no argument, wayland has come as far as to be the default window manager in Ubuntu, and is poised to become X11's replacement in the future. Is waydroid takes advantage of some tricks and advancements in wayland, it is not a downside, but future-proofing.

**Comment by Fuseteam** [comment]
it also does not have flatpak support. i'm starting to think it can't. flatpak simply isn't made for this usecase, nor should it be
see https://github.com/anbox/anbox/issues/62#issuecomment-356383026, note tingping is one of the many flatpak contributors

**Comment by rugk** [comment]
> it also does not have flatpak support.

Well… here we are: https://github.com/waydroid/waydroid/issues/64

But thanks for the pointer that there is an alternative project here… :smiley:

**Comment by Fuseteam** [comment]
Do you guys not read the thread before commenting or something? (no offence)
Half the commenters on both thread just express the wish for flatpak support without looking into what would be required or if it is even feasible

Flatpak and snaps are not comparable just because it was possible with snap does not mean it is possible with flatpaks.

Flatpaks are designed for desktop applications snap is designed for running services headlessly and maintenance free (as that's needed in the IoT field where snaps are also used)

Point in case snap install nextcloud setups a nextcloud server ready to configure via its web interface. Flatpak simply cannot do that, nor does it need to.
